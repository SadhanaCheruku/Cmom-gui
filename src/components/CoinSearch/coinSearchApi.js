import axios from "axios";

export const coinSearchApi = {
  getCoins: async () => {
    try {
      const response = await axios.get(
        "https://api.coingecko.com/api/v3/coins/markets?vs_currency=eur&order=market_cap_desc&per_page=100&page=1&sparkline=false"
      );
      return response.data;
    } catch (error) {
      console.log(error);
      throw error;
    }
  },
};
