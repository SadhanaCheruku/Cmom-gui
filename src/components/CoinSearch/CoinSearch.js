import React, { useState, useEffect } from "react";
import { Box, Grid } from "@material-ui/core";
import Header from "../shared/Header/Header";
import { coinSearchApi } from "./coinSearchApi";
import CustomTable from "../shared/CustomTable/CustomTable";

const columns = [
  { id: "name", label: "Name", minWidth: 100 },
  {
    id: "image",
    label: "Image",
    minWidth: 100,
    align: "right",
    format: (value) => value.toLocaleString("en-US"),
    renderAsImage: true,
  },
  {
    id: "symbol",
    label: "Symbol",
    minWidth: 100,
    align: "right",
    format: (value) => value,
    renderAsImage: false,
  },
  {
    id: "current_price",
    label: "Current Price",
    minWidth: 100,
    align: "right",
    format: (value) => value,
    renderAsImage: false,
  },
  {
    id: "high_24h",
    label: "High 24 hour Price",
    minWidth: 100,
    align: "right",
    format: (value) => value,
    renderAsImage: false,
  },
  {
    id: "low_24h",
    label: "Low 24 hour Price",
    minWidth: 100,
    align: "right",
    format: (value) => value,
    renderAsImage: false,
  },
];

const CoinSearch = () => {
  const [searchData, setSearchData] = useState([]);

  //Since the Cryptocurrency Prices and Market Capitalization data is real time, hence not using store.
  const fetchCoinList = async () => {
    const response = await coinSearchApi.getCoins();
    setSearchData(response);
  };
  useEffect(() => {
    fetchCoinList();
  }, []);

  return (
    <>
      <Header />
      <Grid
        container
        spacing={0}
        direction="column"
        alignItems="center"
        justify="center"
      >
        <Box>
          {
            <CustomTable
              searchResult={searchData}
              columns={columns}
              title={"European Cryptocurrency Prices and Market Capitalization"}
            />
          }
        </Box>
      </Grid>
    </>
  );
};

export default CoinSearch;
