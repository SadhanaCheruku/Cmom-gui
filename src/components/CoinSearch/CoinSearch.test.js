import { render } from "@testing-library/react";
import CoinSearch from "./CoinSearch";
import React from "react";
import axios from "axios";
import { BrowserRouter as Router } from "react-router-dom";

jest.mock("axios");
describe("SearchLibrary", () => {
  it("render component as expected", () => {
    axios.get.mockImplementationOnce(() =>
      Promise.resolve({
        data: [
          {
            id: "bitcoin",
            symbol: "btc",
            name: "Bitcoin",
            image:
              "https://assets.coingecko.com/coins/images/1/large/bitcoin.png?1547033579",
            current_price: 39424,
            market_cap: 729798545595,
            market_cap_rank: 1,
            fully_diluted_valuation: 822788892446,
            total_volume: 56674713770,
            high_24h: 40131,
            low_24h: 37907,
            price_change_24h: 1415.58,
            price_change_percentage_24h: 3.72438,
            market_cap_change_24h: 21531284230,
            market_cap_change_percentage_24h: 3.03999,
            circulating_supply: 18626612,
            total_supply: 21000000,
            max_supply: 21000000,
            ath: 40131,
            ath_change_percentage: -2.50649,
            ath_date: "2021-02-12T00:23:27.963Z",
            atl: 51.3,
            atl_change_percentage: 76170.27488,
            atl_date: "2013-07-05T00:00:00.000Z",
            roi: null,
            last_updated: "2021-02-12T11:30:21.581Z",
          },
          {
            id: "ethereum",
            symbol: "eth",
            name: "Ethereum",
            image:
              "https://assets.coingecko.com/coins/images/279/large/ethereum.png?1595348880",
            current_price: 1471.85,
            market_cap: 168653072707,
            market_cap_rank: 2,
            fully_diluted_valuation: null,
            total_volume: 28561397397,
            high_24h: 1492.25,
            low_24h: 1434.03,
            price_change_24h: 7.43,
            price_change_percentage_24h: 0.50743,
            market_cap_change_24h: 455257520,
            market_cap_change_percentage_24h: 0.27067,
            circulating_supply: 114639697.749,
            total_supply: null,
            max_supply: null,
            ath: 1506.36,
            ath_change_percentage: -3.02005,
            ath_date: "2021-02-10T08:14:41.266Z",
            atl: 0.381455,
            atl_change_percentage: 382871.55746,
            atl_date: "2015-10-20T00:00:00.000Z",
            roi: {
              times: 48.965055381080205,
              currency: "btc",
              percentage: 4896.505538108021,
            },
            last_updated: "2021-02-12T11:32:08.434Z",
          },
        ],
        status: 200,
        statusText: "",
        headers: {
          "cache-control": "max-age=30, public, must-revalidate, s-maxage=30",
          "content-type": "application/json; charset=utf-8",
        },
        config: {
          url:
            "https://api.coingecko.com/api/v3/coins/markets?vs_currency=eur&order=market_cap_desc&per_page=100&page=1&sparkline=false",
          method: "get",
          headers: { Accept: "application/json, text/plain, */*" },
          transformRequest: [null],
          transformResponse: [null],
          timeout: 0,
          xsrfCookieName: "XSRF-TOKEN",
          xsrfHeaderName: "X-XSRF-TOKEN",
          maxContentLength: -1,
        },
        request: {},
      })
    );
    const { asFragment } = render(
      <Router>
        <CoinSearch />
      </Router>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
