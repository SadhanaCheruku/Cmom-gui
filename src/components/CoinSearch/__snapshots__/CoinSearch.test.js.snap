// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchLibrary render component as expected 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
  >
    <header
      class="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary mui-fixed MuiPaper-elevation4"
    >
      <div
        class="MuiToolbar-root MuiToolbar-dense MuiToolbar-gutters"
      >
        <h4
          class="MuiTypography-root MuiTypography-h4 MuiTypography-colorInherit"
        >
          CoinLive
        </h4>
      </div>
    </header>
  </div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
  >
    <div
      class="MuiBox-root MuiBox-root-1"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
      >
        <h5
          class="MuiTypography-root makeStyles-typoGraphy-3 MuiTypography-h5 MuiTypography-colorPrimary"
        >
          European Cryptocurrency Prices and Market Capitalization
        </h5>
        <div
          class="MuiPaper-root makeStyles-root-2 MuiPaper-elevation1 MuiPaper-rounded"
        >
          <div
            class="MuiTableContainer-root"
          >
            <table
              aria-label="sticky table"
              class="MuiTable-root MuiTable-stickyHeader"
            >
              <thead
                class="MuiTableHead-root"
              >
                <tr
                  class="MuiTableRow-root MuiTableRow-head"
                >
                  <th
                    class="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader"
                    scope="col"
                    style="min-width: 100px;"
                  >
                    Name
                  </th>
                  <th
                    class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-stickyHeader"
                    scope="col"
                    style="min-width: 100px;"
                  >
                    Image
                  </th>
                  <th
                    class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-stickyHeader"
                    scope="col"
                    style="min-width: 100px;"
                  >
                    Symbol
                  </th>
                  <th
                    class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-stickyHeader"
                    scope="col"
                    style="min-width: 100px;"
                  >
                    Current Price
                  </th>
                  <th
                    class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-stickyHeader"
                    scope="col"
                    style="min-width: 100px;"
                  >
                    High 24 hour Price
                  </th>
                  <th
                    class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-stickyHeader"
                    scope="col"
                    style="min-width: 100px;"
                  >
                    Low 24 hour Price
                  </th>
                </tr>
              </thead>
              <tbody
                class="MuiTableBody-root"
              />
            </table>
          </div>
          <div
            class="MuiTablePagination-root"
          >
            <div
              class="MuiToolbar-root MuiToolbar-regular MuiTablePagination-toolbar MuiToolbar-gutters"
            >
              <div
                class="MuiTablePagination-spacer"
              />
              <p
                class="MuiTypography-root MuiTablePagination-caption MuiTypography-body2 MuiTypography-colorInherit"
                id="mui-34995"
              >
                Rows per page:
              </p>
              <div
                class="MuiInputBase-root MuiTablePagination-input MuiTablePagination-selectRoot"
              >
                <div
                  aria-haspopup="listbox"
                  aria-labelledby="mui-34995 mui-34922"
                  class="MuiSelect-root MuiSelect-select MuiTablePagination-select MuiSelect-selectMenu MuiInputBase-input"
                  id="mui-34922"
                  role="button"
                  tabindex="0"
                >
                  10
                </div>
                <input
                  aria-hidden="true"
                  class="MuiSelect-nativeInput"
                  tabindex="-1"
                  value="10"
                />
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSelect-icon MuiTablePagination-selectIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 10l5 5 5-5z"
                  />
                </svg>
              </div>
              <p
                class="MuiTypography-root MuiTablePagination-caption MuiTypography-body2 MuiTypography-colorInherit"
              >
                0-0 of 0
              </p>
              <div
                class="MuiTablePagination-actions"
              >
                <button
                  aria-label="Previous page"
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit Mui-disabled Mui-disabled"
                  disabled=""
                  tabindex="-1"
                  title="Previous page"
                  type="button"
                >
                  <span
                    class="MuiIconButton-label"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z"
                      />
                    </svg>
                  </span>
                </button>
                <button
                  aria-label="Next page"
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit Mui-disabled Mui-disabled"
                  disabled=""
                  tabindex="-1"
                  title="Next page"
                  type="button"
                >
                  <span
                    class="MuiIconButton-label"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
                      />
                    </svg>
                  </span>
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
